_rules/7d6734.md

```diff
@@ -0,0 +1,183 @@
+---
+id: 7d6734
+name: SVG image has accessible name
+rule_type: atomic
+description: |
+   This rule checks that each SVG image element that is included in the accessibility tree has an accessible name.
+accessibility_requirements:
+  wcag20:1.1.1: # Non-Text Content (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+authors:
+  - Bryn Anderson
+---
+
+## Applicability
+
+The rule applies to any element in the [SVG](https://www.w3.org/2000/svg) namespace with a [semantic role](#semantic-role) of either `img`, `graphics-document`, `graphics-object`, `graphics-symbol`, that is [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+**Note**: The [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0/#include_elements) specifies that many elements in the SVG namespace are purely presentational and should not be included in the accessibility tree unless indicated otherwise through the use of text alternatve content and WAI ARIA attributes.
```

Comment:
Thanks for this @carlosapaduarte .

I also received some clarity from Leonie Watson basically agreeing. To be fair to @WilcoFiers he had suggested I update the applicability from "semantic role" to "explicit semantic role". This would render these two test cases inapplicable.

Reply:
Having thought about this again (again again) I don't think the applicability should be restricted to elements with an explicit role because that wouldn't catch the following which should be a fail:

```
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
  <title> </title>
	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
</svg>
```

The role whether implicit or not only part determines if the element should be included in the accessibility tree.

This means the following:

**Inapplicable** <svg> element renders a <circle> element to the screen but the <circle> element has no semantic meaning so it is not included in the a11y tree.
```html
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
  	<circle  cx="50" cy="50" r="40" fill="yellow"></circle>
</svg>
```

**Fail** <svg> element renders a <circle> element to the screen and the <circle> element has a role of graphics-symbol which includes it in the a11y tree.
```html
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
  	<circle role="graphics-symbol"  cx="50" cy="50" r="40" fill="yellow"></circle>
</svg>
```

In plain language... this rule should target SVG image elements that are visible on screen to the user and that are included in the accessibility tree.